### PR TITLE
Improved build

### PR DESCRIPTION
**Pull request by syrusakbary**
Objectives of this PR:
* [x] Have build to work outside of a travis environment (so it can be run easily in `osx` or `linux` by just executing `./Build/build.sh`)
* [x] Improve documentation showcasing `./Build/build.sh` as a local build script
* [ ] Improve build times for Travis 
  * [x] Use smart-caching of `llvm6` directory (only downloading llvm if not present)
  * [ ] Use `ccache` when possible, so subsequent travis builds take much less time to execute

### CODE REVIEW

**Comment by syrusakbary** [comment]
As a note, `build.sh` and `travis-build.sh` are almost identical, the only difference is:
* llvm will only be downloaded if directory is non-existing
* `release` and `debug` dirs will only be created if they are unexisting (so mkdir doesn't fail)
* Added default values for some env variables (if not present) (so build script works out of the box).

File: README.md [hunk 1]
```diff
@@ -55,15 +59,15 @@ Usage: wavm [switches] [programfile] [--] [arguments]
...
 
 ```
 wavm Examples/helloworld.wast
 wavm Examples/Benchmark/Benchmark.wast
 wavm Examples/zlib.wast
 ```
 
-WebAssembly programs that export a main function with the standard parameters will be passed in the command line arguments.  If the same main function returns a i32 type it will become the exit code.  WAVM supports Emscripten's defined I/O functions so programs can read from stdin and write to stdout and stderr.  See [echo.wast](Examples/echo.wast) for an example of a program that echos the command line arguments back out through stdout.
+WebAssembly programs that export a main function with the standard parameters will be passed in the command line arguments. If the same main function returns a i32 type it will become the exit code. WAVM supports Emscripten's defined I/O functions so programs can read from stdin and write to stdout and stderr. See [echo.wast](Examples/echo.wast) for an example of a program that echos the command line arguments back out through stdout.
```
Comment (syrusakbary) [inline README.md]: These changes are done automatically by `prettier` (markdown)

**Comment by AndrewScheidecker** [comment]
> Have build to work outside of a travis environment (so it can be run easily in osx or linux by just executing ./Build/build.sh)
> Improve documentation showcasing ./Build/build.sh as a local build script

I think maybe a better way to do this would be to make a `configure.sh` that just downloads LLVM and runs `cmake`, and leaves the `make` to the user.

It's worth noting that the comment in `travis-build.sh` was out of date: I think the LLVM6 apt repo is whitelisted now on Travis, so downloading a tarball isn't necessary. However, I think it might be necessary to use a fork of LLVM at some point, so I didn't want to simplify that and then need to bring it back.

I'm also considering moving to Azure Pipelines for non-Windows CI: it supports Windows as well, and has a much more generous free plan for open-source projects (10 parallel jobs, and no limit on how long a job can take). The lack of a limit on job execution time means that it's possible to build LLVM from source, should it be necessary (i.e. if WAVM forks LLVM, or uses a pre-release version).

> Use smart-caching of llvm6 directory (only downloading llvm if not present)

I doubt this will make a big difference, since it's just a difference of which server it downloads a tarball from.

> Improve build times for Travis 

I should say that I think the Travis latency is actually not too bad right now. I'd see improving build times on Travis as an opportunity to add more configurations to test.

A bottleneck on Travis is the time to install all the sources from apt. IIRC that's more than half of each job.

However, as I'm considering switching to Azure Pipelines for all platforms, I wouldn't put much effort into trying to make Travis faster right now.

**Comment by syrusakbary** [comment]
Thanks for the feedback! Closing the PR for now :)

**Pull request closed without merging**